Build fails with GCC 16

Hi there,

jsoncpp is currently failing to build with GCC 16 on GNU/Linux:

```
2026/05/06 03:36:53 INFO [1/18] /usr/local/bin/c++ -DJSONCPP_HAS_STRING_VIEW=1 -DJSON_DLL -I/home/build/include -I/home/build/src/lib_json/../../include -O3 -DNDEBUG -Wall -Wconversion -Wshadow -Wextra -Wpedantic -MD -MT src/test_lib_json/CMakeFiles/jsoncpp_test.dir/main.cpp.o -MF src/test_lib_json/CMakeFiles/jsoncpp_test.dir/main.cpp.o.d -o src/test_lib_json/CMakeFiles/jsoncpp_test.dir/main.cpp.o -c /home/build/src/test_lib_json/main.cpp
2026/05/06 03:36:53 INFO [2/18] /usr/local/bin/c++ -DJSONCPP_HAS_STRING_VIEW=1 -DJSON_DLL -I/home/build/include -I/home/build/src/lib_json/../../include -O3 -DNDEBUG -Wall -Wconversion -Wshadow -Wextra -Wpedantic -MD -MT src/test_lib_json/CMakeFiles/jsoncpp_test.dir/fuzz.cpp.o -MF src/test_lib_json/CMakeFiles/jsoncpp_test.dir/fuzz.cpp.o.d -o src/test_lib_json/CMakeFiles/jsoncpp_test.dir/fuzz.cpp.o -c /home/build/src/test_lib_json/fuzz.cpp
2026/05/06 03:36:53 INFO [3/18] /usr/local/bin/c++ -DJSONCPP_HAS_STRING_VIEW=1 -DJSON_DLL -I/home/build/include -I/home/build/src/lib_json/../../include -O3 -DNDEBUG -Wall -Wconversion -Wshadow -Wextra -Wpedantic -MD -MT src/test_lib_json/CMakeFiles/jsoncpp_test.dir/jsontest.cpp.o -MF src/test_lib_json/CMakeFiles/jsoncpp_test.dir/jsontest.cpp.o.d -o src/test_lib_json/CMakeFiles/jsoncpp_test.dir/jsontest.cpp.o -c /home/build/src/test_lib_json/jsontest.cpp
2026/05/06 03:36:54 INFO [4/18] /usr/local/bin/c++ -DJSONCPP_HAS_STRING_VIEW=1 -DJSON_DLL -I/home/build/include -I/home/build/src/lib_json/../../include -O3 -DNDEBUG -Wall -Wconversion -Wshadow -Wextra -Wpedantic -MD -MT src/jsontestrunner/CMakeFiles/jsontestrunner_exe.dir/main.cpp.o -MF src/jsontestrunner/CMakeFiles/jsontestrunner_exe.dir/main.cpp.o.d -o src/jsontestrunner/CMakeFiles/jsontestrunner_exe.dir/main.cpp.o -c /home/build/src/jsontestrunner/main.cpp
2026/05/06 03:36:55 WARN ninja: job failed: /usr/local/bin/c++ -DJSONCPP_HAS_STRING_VIEW=1 -DJSON_DLL -I/home/build/include -I/home/build/src/lib_json/../../include -O3 -DNDEBUG -Wall -Wconversion -Wshadow -Wextra -Wpedantic -MD -MT src/test_lib_json/CMakeFiles/jsoncpp_test.dir/main.cpp.o -MF src/test_lib_json/CMakeFiles/jsoncpp_test.dir/main.cpp.o.d -o src/test_lib_json/CMakeFiles/jsoncpp_test.dir/main.cpp.o -c /home/build/src/test_lib_json/main.cpp
2026/05/06 03:36:55 INFO /home/build/src/test_lib_json/main.cpp: In member function 'virtual void TestValueTestWideString::runTestCase()':
2026/05/06 03:36:55 INFO /home/build/src/test_lib_json/main.cpp:1996:27: error: conversion from 'const char8_t [10]' to non-scalar type 'const std::string' {aka 'const std::__cxx11::basic_string<char>'} requested
2026/05/06 03:36:55 INFO  1996 |   const std::string uni = u8"\u5f0f\uff0c\u8fdb"; // "式，进"
2026/05/06 03:36:55 INFO       |                           ^~~~~~~~~~~~~~~~~~~~~~
2026/05/06 03:36:55 INFO /home/build/src/test_lib_json/main.cpp: In lambda function:
2026/05/06 03:36:55 INFO /home/build/src/test_lib_json/main.cpp:3605:12: warning: implicit capture of 'this' via '[=]' is deprecated in C++20 [-Wdeprecated]
2026/05/06 03:36:55 INFO  3605 |     return [=](const Value& root) { JSONTEST_ASSERT_EQUAL(root, v); };
2026/05/06 03:36:55 INFO       |            ^
2026/05/06 03:36:55 INFO /home/build/src/test_lib_json/main.cpp:3605:12: note: add explicit 'this' or '*this' capture
2026/05/06 03:36:55 INFO In file included from /home/build/src/test_lib_json/main.cpp:14:
2026/05/06 03:36:55 INFO /home/build/src/test_lib_json/jsontest.h: In instantiation of 'JsonTest::TestResult& JsonTest::checkEqual(TestResult&, T, U, const char*, unsigned int, const char*) [with T = const char8_t*; U = std::__cxx11::basic_string<char>]':
2026/05/06 03:36:55 INFO /home/build/src/test_lib_json/main.cpp:3226:5:   required from here
2026/05/06 03:36:55 INFO   222 |   JsonTest::checkEqual(*result_, expected, actual, __FILE__, __LINE__,         \
2026/05/06 03:36:55 INFO       |   ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2026/05/06 03:36:55 INFO   223 |                        #expected " == " #actual)
2026/05/06 03:36:55 INFO       |                        ~~~~~~~~~~~~~~~~~~~~~~~~~
2026/05/06 03:36:55 INFO /home/build/src/test_lib_json/jsontest.h:178:7: error: no matching function for call to 'std::__cxx11::basic_string<char>::basic_string(const char8_t*&)'
2026/05/06 03:36:55 INFO   178 |   if (static_cast<U>(expected) != actual) {
2026/05/06 03:36:55 INFO       |       ^~~~~~~~~~~~~~~~~~~~~~~~
2026/05/06 03:36:55 INFO   • there are 17 candidates
2026/05/06 03:36:55 INFO In file included from /usr/include/c++/16/string:58,
2026/05/06 03:36:55 INFO                  from /usr/include/c++/16/bits/locale_classes.h:43,
2026/05/06 03:36:55 INFO                  from /usr/include/c++/16/bits/ios_base.h:43,
2026/05/06 03:36:55 INFO                  from /usr/include/c++/16/iomanip:47,
2026/05/06 03:36:55 INFO                  from /home/build/src/test_lib_json/jsontest.h:11:
2026/05/06 03:36:55 INFO     • candidate 1: 'template<class _Tp, class> constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _Tp&, const _Alloc&) [with <template-parameter-2-2> = _Tp; _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:911:9:
2026/05/06 03:36:55 INFO         911 |         basic_string(const _Tp& __t, const _Alloc& __a = _Alloc())
2026/05/06 03:36:55 INFO             |         ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • template argument deduction/substitution failed:
2026/05/06 03:36:55 INFO In file included from /usr/include/c++/16/bits/cpp_type_traits.h:42,
2026/05/06 03:36:55 INFO                  from /usr/include/c++/16/bits/stl_algobase.h:60,
2026/05/06 03:36:55 INFO                  from /usr/include/c++/16/deque:65,
2026/05/06 03:36:55 INFO                  from /home/build/src/test_lib_json/jsontest.h:10:
2026/05/06 03:36:55 INFO         • /usr/include/c++/16/type_traits: In substitution of 'template<bool _Cond, class _Tp> using std::enable_if_t = typename std::enable_if::type [with bool _Cond = false; _Tp = void]':
2026/05/06 03:36:55 INFO         • required by substitution of 'template<class _CharT, class _Traits, class _Alloc> template<class _Tp, class _Res> using std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::_If_sv = std::enable_if_t<((bool)std::__and_<std::is_convertible<const _Tp&, std::basic_string_view<_CharT, _Traits> >, std::__not_<std::is_convertible<const _Tp*, const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>*> >, std::__not_<std::is_convertible<const _Tp&, const _CharT*> > >::value), _Res> [with _Tp = const char8_t*; _Res = void; _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO           /usr/include/c++/16/bits/basic_string.h:159:8:   
2026/05/06 03:36:55 INFO             159 |         using _If_sv = enable_if_t<
2026/05/06 03:36:55 INFO                 |               ^~~~~~
2026/05/06 03:36:55 INFO         • required from 'JsonTest::TestResult& JsonTest::checkEqual(TestResult&, T, U, const char*, unsigned int, const char*) [with T = const char8_t*; U = std::__cxx11::basic_string<char>]'
2026/05/06 03:36:55 INFO           /usr/include/c++/16/bits/basic_string.h:908:30:   
2026/05/06 03:36:55 INFO             908 |       template<typename _Tp, typename = _If_sv<_Tp, void>>
2026/05/06 03:36:55 INFO                 |                              ^~~~~~~~
2026/05/06 03:36:55 INFO         • required from here
2026/05/06 03:36:55 INFO           /home/build/src/test_lib_json/main.cpp:3226:5:   
2026/05/06 03:36:55 INFO             222 |   JsonTest::checkEqual(*result_, expected, actual, __FILE__, __LINE__,         \
2026/05/06 03:36:55 INFO                 |   ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2026/05/06 03:36:55 INFO             223 |                        #expected " == " #actual)
2026/05/06 03:36:55 INFO                 |                        ~~~~~~~~~~~~~~~~~~~~~~~~~
2026/05/06 03:36:55 INFO         • error: no type named 'type' in 'struct std::enable_if<false, void>'
2026/05/06 03:36:55 INFO           /usr/include/c++/16/type_traits:2942:11:
2026/05/06 03:36:55 INFO            2942 |     using enable_if_t = typename enable_if<_Cond, _Tp>::type;
2026/05/06 03:36:55 INFO                 |           ^~~~~~~~~~~
2026/05/06 03:36:55 INFO     • /home/build/src/test_lib_json/jsontest.h: In instantiation of 'JsonTest::TestResult& JsonTest::checkEqual(TestResult&, T, U, const char*, unsigned int, const char*) [with T = const char8_t*; U = std::__cxx11::basic_string<char>]':
2026/05/06 03:36:55 INFO     • required from here
2026/05/06 03:36:55 INFO       /home/build/src/test_lib_json/main.cpp:3226:5:   
2026/05/06 03:36:55 INFO         222 |   JsonTest::checkEqual(*result_, expected, actual, __FILE__, __LINE__,         \
2026/05/06 03:36:55 INFO             |   ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2026/05/06 03:36:55 INFO         223 |                        #expected " == " #actual)
2026/05/06 03:36:55 INFO             |                        ~~~~~~~~~~~~~~~~~~~~~~~~~
2026/05/06 03:36:55 INFO     • candidate 2: 'template<class _Tp, class> constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _Tp&, size_type, size_type, const _Alloc&) [with <template-parameter-2-2> = _Tp; _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:899:9:
2026/05/06 03:36:55 INFO         899 |         basic_string(const _Tp& __t, size_type __pos, size_type __n,
2026/05/06 03:36:55 INFO             |         ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 3 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 3: 'template<class _InputIterator, class> constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&) [with <template-parameter-2-2> = _InputIterator; _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:876:9:
2026/05/06 03:36:55 INFO         876 |         basic_string(_InputIterator __beg, _InputIterator __end,
2026/05/06 03:36:55 INFO             |         ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 2 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 4: 'template<class> constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(size_type, _CharT, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:753:7:
2026/05/06 03:36:55 INFO         753 |       basic_string(size_type __n, _CharT __c, const _Alloc& __a = _Alloc())
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 2 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 5: 'template<class> constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _CharT*, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:730:7:
2026/05/06 03:36:55 INFO         730 |       basic_string(const _CharT* __s, const _Alloc& __a = _Alloc())
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • template argument deduction/substitution failed:
2026/05/06 03:36:55 INFO         •   cannot convert 'expected' (type 'const char8_t*') to type 'const char*'
2026/05/06 03:36:55 INFO           /home/build/src/test_lib_json/jsontest.h:178:22:
2026/05/06 03:36:55 INFO             178 |   if (static_cast<U>(expected) != actual) {
2026/05/06 03:36:55 INFO                 |                      ^~~~~~~~
2026/05/06 03:36:55 INFO     • candidate 6: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&&, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:832:7:
2026/05/06 03:36:55 INFO         832 |       basic_string(basic_string&& __str, const _Alloc& __a)
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 2 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 7: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:827:7:
2026/05/06 03:36:55 INFO         827 |       basic_string(const basic_string& __str, const _Alloc& __a)
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 2 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 8: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(std::initializer_list<_Tp>, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:822:7:
2026/05/06 03:36:55 INFO         822 |       basic_string(initializer_list<_CharT> __l, const _Alloc& __a = _Alloc())
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • no known conversion for argument 1 from 'const char8_t*' to 'std::initializer_list<char>'
2026/05/06 03:36:55 INFO         /usr/include/c++/16/bits/basic_string.h:822:45:
2026/05/06 03:36:55 INFO           822 |       basic_string(initializer_list<_CharT> __l, const _Alloc& __a = _Alloc())
2026/05/06 03:36:55 INFO               |                    ~~~~~~~~~~~~~~~~~~~~~~~~~^~~
2026/05/06 03:36:55 INFO     • candidate 9: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:766:7:
2026/05/06 03:36:55 INFO         766 |       basic_string(basic_string&& __str) noexcept
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • no known conversion for argument 1 from 'const char8_t*' to 'std::__cxx11::basic_string<char>&&'
2026/05/06 03:36:55 INFO         /usr/include/c++/16/bits/basic_string.h:766:35:
2026/05/06 03:36:55 INFO           766 |       basic_string(basic_string&& __str) noexcept
2026/05/06 03:36:55 INFO               |                    ~~~~~~~~~~~~~~~^~~~~
2026/05/06 03:36:55 INFO     • candidate 10: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _CharT*, size_type, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; size_type = long unsigned int]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:708:7:
2026/05/06 03:36:55 INFO         708 |       basic_string(const _CharT* __s, size_type __n,
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 3 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 11: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, size_type, size_type, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; size_type = long unsigned int]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:668:7:
2026/05/06 03:36:55 INFO         668 |       basic_string(const basic_string& __str, size_type __pos,
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 4 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 12: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, size_type, size_type) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; size_type = long unsigned int]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:650:7:
2026/05/06 03:36:55 INFO         650 |       basic_string(const basic_string& __str, size_type __pos,
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 3 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 13: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, size_type, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; size_type = long unsigned int]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:633:7:
2026/05/06 03:36:55 INFO         633 |       basic_string(const basic_string& __str, size_type __pos,
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 3 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 14: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:617:7:
2026/05/06 03:36:55 INFO         617 |       basic_string(const basic_string& __str)
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • no known conversion for argument 1 from 'const char8_t*' to 'const std::__cxx11::basic_string<char>&'
2026/05/06 03:36:55 INFO         /usr/include/c++/16/bits/basic_string.h:617:40:
2026/05/06 03:36:55 INFO           617 |       basic_string(const basic_string& __str)
2026/05/06 03:36:55 INFO               |                    ~~~~~~~~~~~~~~~~~~~~^~~~~
2026/05/06 03:36:55 INFO     • candidate 15: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:605:7:
2026/05/06 03:36:55 INFO         605 |       basic_string(const _Alloc& __a) _GLIBCXX_NOEXCEPT
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • no known conversion for argument 1 from 'const char8_t*' to 'const std::allocator<char>&'
2026/05/06 03:36:55 INFO         /usr/include/c++/16/bits/basic_string.h:605:34:
2026/05/06 03:36:55 INFO           605 |       basic_string(const _Alloc& __a) _GLIBCXX_NOEXCEPT
2026/05/06 03:36:55 INFO               |                    ~~~~~~~~~~~~~~^~~
2026/05/06 03:36:55 INFO     • candidate 16: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string() requires  is_default_constructible_v<_Alloc> [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:589:7:
2026/05/06 03:36:55 INFO         589 |       basic_string()
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 0 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 17: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(__sv_wrapper, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:191:7:
2026/05/06 03:36:55 INFO         191 |       basic_string(__sv_wrapper __svw, const _Alloc& __a)
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 2 arguments, 1 provided
2026/05/06 03:36:55 INFO /home/build/src/test_lib_json/jsontest.h:180:31: error: no matching function for call to 'std::__cxx11::basic_string<char>::basic_string(const char8_t*&)'
2026/05/06 03:36:55 INFO   180 |     result << "Expected: " << static_cast<U>(expected) << "\n";
2026/05/06 03:36:55 INFO       |                               ^~~~~~~~~~~~~~~~~~~~~~~~
2026/05/06 03:36:55 INFO   • there are 17 candidates
2026/05/06 03:36:55 INFO     • candidate 1: 'template<class _Tp, class> constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _Tp&, const _Alloc&) [with <template-parameter-2-2> = _Tp; _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:911:9:
2026/05/06 03:36:55 INFO         911 |         basic_string(const _Tp& __t, const _Alloc& __a = _Alloc())
2026/05/06 03:36:55 INFO             |         ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • template argument deduction/substitution failed:
2026/05/06 03:36:55 INFO     • candidate 2: 'template<class _Tp, class> constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _Tp&, size_type, size_type, const _Alloc&) [with <template-parameter-2-2> = _Tp; _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:899:9:
2026/05/06 03:36:55 INFO         899 |         basic_string(const _Tp& __t, size_type __pos, size_type __n,
2026/05/06 03:36:55 INFO             |         ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 3 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 3: 'template<class _InputIterator, class> constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&) [with <template-parameter-2-2> = _InputIterator; _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:876:9:
2026/05/06 03:36:55 INFO         876 |         basic_string(_InputIterator __beg, _InputIterator __end,
2026/05/06 03:36:55 INFO             |         ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 2 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 4: 'template<class> constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(size_type, _CharT, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:753:7:
2026/05/06 03:36:55 INFO         753 |       basic_string(size_type __n, _CharT __c, const _Alloc& __a = _Alloc())
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 2 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 5: 'template<class> constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _CharT*, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:730:7:
2026/05/06 03:36:55 INFO         730 |       basic_string(const _CharT* __s, const _Alloc& __a = _Alloc())
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • template argument deduction/substitution failed:
2026/05/06 03:36:55 INFO         •   cannot convert 'expected' (type 'const char8_t*') to type 'const char*'
2026/05/06 03:36:55 INFO           /home/build/src/test_lib_json/jsontest.h:180:46:
2026/05/06 03:36:55 INFO             180 |     result << "Expected: " << static_cast<U>(expected) << "\n";
2026/05/06 03:36:55 INFO                 |                                              ^~~~~~~~
2026/05/06 03:36:55 INFO     • candidate 6: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&&, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:832:7:
2026/05/06 03:36:55 INFO         832 |       basic_string(basic_string&& __str, const _Alloc& __a)
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 2 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 7: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:827:7:
2026/05/06 03:36:55 INFO         827 |       basic_string(const basic_string& __str, const _Alloc& __a)
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 2 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 8: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(std::initializer_list<_Tp>, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:822:7:
2026/05/06 03:36:55 INFO         822 |       basic_string(initializer_list<_CharT> __l, const _Alloc& __a = _Alloc())
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • no known conversion for argument 1 from 'const char8_t*' to 'std::initializer_list<char>'
2026/05/06 03:36:55 INFO         /usr/include/c++/16/bits/basic_string.h:822:45:
2026/05/06 03:36:55 INFO           822 |       basic_string(initializer_list<_CharT> __l, const _Alloc& __a = _Alloc())
2026/05/06 03:36:55 INFO               |                    ~~~~~~~~~~~~~~~~~~~~~~~~~^~~
2026/05/06 03:36:55 INFO     • candidate 9: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:766:7:
2026/05/06 03:36:55 INFO         766 |       basic_string(basic_string&& __str) noexcept
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • no known conversion for argument 1 from 'const char8_t*' to 'std::__cxx11::basic_string<char>&&'
2026/05/06 03:36:55 INFO         /usr/include/c++/16/bits/basic_string.h:766:35:
2026/05/06 03:36:55 INFO           766 |       basic_string(basic_string&& __str) noexcept
2026/05/06 03:36:55 INFO               |                    ~~~~~~~~~~~~~~~^~~~~
2026/05/06 03:36:55 INFO     • candidate 10: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _CharT*, size_type, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; size_type = long unsigned int]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:708:7:
2026/05/06 03:36:55 INFO         708 |       basic_string(const _CharT* __s, size_type __n,
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 3 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 11: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, size_type, size_type, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; size_type = long unsigned int]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:668:7:
2026/05/06 03:36:55 INFO         668 |       basic_string(const basic_string& __str, size_type __pos,
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 4 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 12: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, size_type, size_type) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; size_type = long unsigned int]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:650:7:
2026/05/06 03:36:55 INFO         650 |       basic_string(const basic_string& __str, size_type __pos,
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 3 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 13: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, size_type, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; size_type = long unsigned int]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:633:7:
2026/05/06 03:36:55 INFO         633 |       basic_string(const basic_string& __str, size_type __pos,
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 3 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 14: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:617:7:
2026/05/06 03:36:55 INFO         617 |       basic_string(const basic_string& __str)
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • no known conversion for argument 1 from 'const char8_t*' to 'const std::__cxx11::basic_string<char>&'
2026/05/06 03:36:55 INFO         /usr/include/c++/16/bits/basic_string.h:617:40:
2026/05/06 03:36:55 INFO           617 |       basic_string(const basic_string& __str)
2026/05/06 03:36:55 INFO               |                    ~~~~~~~~~~~~~~~~~~~~^~~~~
2026/05/06 03:36:55 INFO     • candidate 15: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:605:7:
2026/05/06 03:36:55 INFO         605 |       basic_string(const _Alloc& __a) _GLIBCXX_NOEXCEPT
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • no known conversion for argument 1 from 'const char8_t*' to 'const std::allocator<char>&'
2026/05/06 03:36:55 INFO         /usr/include/c++/16/bits/basic_string.h:605:34:
2026/05/06 03:36:55 INFO           605 |       basic_string(const _Alloc& __a) _GLIBCXX_NOEXCEPT
2026/05/06 03:36:55 INFO               |                    ~~~~~~~~~~~~~~^~~
2026/05/06 03:36:55 INFO     • candidate 16: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string() requires  is_default_constructible_v<_Alloc> [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:589:7:
2026/05/06 03:36:55 INFO         589 |       basic_string()
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 0 arguments, 1 provided
2026/05/06 03:36:55 INFO     • candidate 17: 'constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(__sv_wrapper, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]'
2026/05/06 03:36:55 INFO       /usr/include/c++/16/bits/basic_string.h:191:7:
2026/05/06 03:36:55 INFO         191 |       basic_string(__sv_wrapper __svw, const _Alloc& __a)
2026/05/06 03:36:55 INFO             |       ^~~~~~~~~~~~
2026/05/06 03:36:55 INFO       • candidate expects 2 arguments, 1 provided
2026/05/06 03:36:55 INFO /home/build/src/test_lib_json/main.cpp: In instantiation of 'void ReaderTest::checkParse(Input&&, const std::vector<Json::Reader::StructuredError>&) [with Input = const char8_t (&)[30]]':
2026/05/06 03:36:55 INFO /home/build/src/test_lib_json/main.cpp:2976:15:   required from 'void ReaderTest::checkParse(Input&&, const std::vector<Json::Reader::StructuredError>&, const std::string&) [with Input = const char8_t (&)[30]; std::string = std::__cxx11::basic_string<char>]'
2026/05/06 03:36:55 INFO  2976 |     checkParse(input, structured);
2026/05/06 03:36:55 INFO       |     ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
2026/05/06 03:36:55 INFO /home/build/src/test_lib_json/main.cpp:3112:13:   required from here
2026/05/06 03:36:55 INFO  3112 |   checkParse(R"({ "pr)"
2026/05/06 03:36:55 INFO       |   ~~~~~~~~~~^~~~~~~~~~~
2026/05/06 03:36:55 INFO  3113 |              u8"\u4f50\u85e4" // 佐藤
2026/05/06 03:36:55 INFO       |              ~~~~~~~~~~~~~~~~~~~~~~~~
2026/05/06 03:36:55 INFO  3114 |              R"(erty" :: "value" })",
2026/05/06 03:36:55 INFO       |              ~~~~~~~~~~~~~~~~~~~~~~~~
2026/05/06 03:36:55 INFO  3115 |              {{18, 19, "Syntax error: value, object or array expected."}},
2026/05/06 03:36:55 INFO       |              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2026/05/06 03:36:55 INFO  3116 |              "* Line 1, Column 19\n  Syntax error: value, object or array "
2026/05/06 03:36:55 INFO       |              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2026/05/06 03:36:55 INFO  3117 |              "expected.\n");
2026/05/06 03:36:55 INFO       |              ~~~~~~~~~~~~~~
2026/05/06 03:36:55 INFO /home/build/src/test_lib_json/main.cpp:2968:35: error: no matching function for call to 'Json::Reader::parse(const char8_t [30], Json::Value&)'
2026/05/06 03:36:55 INFO  2968 |     JSONTEST_ASSERT(!reader->parse(input, root));
2026/05/06 03:36:55 INFO       |                      ~~~~~~~~~~~~~^~~~~~~~~~~~~
2026/05/06 03:36:55 INFO /home/build/src/test_lib_json/jsontest.h:202:7: note: in definition of macro 'JSONTEST_ASSERT'
2026/05/06 03:36:55 INFO   202 |   if (expr) {                                                                  \
2026/05/06 03:36:55 INFO       |       ^~~~
2026/05/06 03:36:55 INFO   • there are 3 candidates
2026/05/06 03:36:55 INFO   • in definition of macro 'JSONTEST_ASSERT'
2026/05/06 03:36:55 INFO       202 |   if (expr) {                                                                  \
2026/05/06 03:36:55 INFO           |       ^~~~
2026/05/06 03:36:55 INFO In file included from /home/build/include/json/json.h:11,
2026/05/06 03:36:55 INFO                  from /home/build/src/test_lib_json/main.cpp:23:
2026/05/06 03:36:55 INFO     • candidate 1: 'bool Json::Reader::parse(Json::IStream&, Json::Value&, bool)'
2026/05/06 03:36:55 INFO       /home/build/include/json/reader.h:104:8:
2026/05/06 03:36:55 INFO         104 |   bool parse(IStream& is, Value& root, bool collectComments = true);
2026/05/06 03:36:55 INFO             |        ^~~~~
2026/05/06 03:36:55 INFO       • no known conversion for argument 1 from 'const char8_t [30]' to 'Json::IStream&' {aka 'std::basic_istream<char>&'}
2026/05/06 03:36:55 INFO         /home/build/include/json/reader.h:104:23:
2026/05/06 03:36:55 INFO           104 |   bool parse(IStream& is, Value& root, bool collectComments = true);
2026/05/06 03:36:55 INFO               |              ~~~~~~~~~^~
2026/05/06 03:36:55 INFO     • candidate 2: 'bool Json::Reader::parse(const std::string&, Json::Value&, bool)'
2026/05/06 03:36:55 INFO       /home/build/include/json/reader.h:77:8:
2026/05/06 03:36:55 INFO          77 |   bool parse(const std::string& document, Value& root,
2026/05/06 03:36:55 INFO             |        ^~~~~
2026/05/06 03:36:55 INFO       • no known conversion for argument 1 from 'const char8_t [30]' to 'const std::string&' {aka 'const std::__cxx11::basic_string<char>&'}
2026/05/06 03:36:55 INFO         /home/build/include/json/reader.h:77:33:
2026/05/06 03:36:55 INFO            77 |   bool parse(const std::string& document, Value& root,
2026/05/06 03:36:55 INFO               |              ~~~~~~~~~~~~~~~~~~~^~~~~~~~
2026/05/06 03:36:55 INFO     • candidate 3: 'bool Json::Reader::parse(const char*, const char*, Json::Value&, bool)'
2026/05/06 03:36:55 INFO       /home/build/include/json/reader.h:99:8:
2026/05/06 03:36:55 INFO          99 |   bool parse(const char* beginDoc, const char* endDoc, Value& root,
2026/05/06 03:36:55 INFO             |        ^~~~~
2026/05/06 03:36:55 INFO       • candidate expects 4 arguments, 2 provided
2026/05/06 03:36:58 WARN ninja: subcommand failed
```

Please let me know if you need more information about the build.

Thanks!

--- a/.github/workflows/cmake.yml
+++ b/.github/workflows/cmake.yml
@@ -12,7 +12,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, windows-latest, macos-latest]
-        cxx_standard: [11, 17]
+        cxx_standard: [11, 17, 20]
         
     steps:
       - name: checkout project

--- a/src/test_lib_json/main.cpp
+++ b/src/test_lib_json/main.cpp
@@ -1993,7 +1993,7 @@ JSONTEST_FIXTURE_LOCAL(ValueTest, StaticString) {
 
 JSONTEST_FIXTURE_LOCAL(ValueTest, WideString) {
   // https://github.com/open-source-parsers/jsoncpp/issues/756
-  const std::string uni = u8"\u5f0f\uff0c\u8fdb"; // "式，进"
+  const std::string uni = reinterpret_cast<const char*>(u8"\u5f0f\uff0c\u8fdb"); // "式，进"
   std::string styled;
   {
     Json::Value v;
@@ -3109,9 +3109,9 @@ JSONTEST_FIXTURE_LOCAL(ReaderTest, strictModeParseNumber) {
 }
 
 JSONTEST_FIXTURE_LOCAL(ReaderTest, parseChineseWithOneError) {
-  checkParse(R"({ "pr)"
+  checkParse(reinterpret_cast<const char*>(R"({ "pr)"
              u8"\u4f50\u85e4" // 佐藤
-             R"(erty" :: "value" })",
+             R"(erty" :: "value" })"),
              {{18, 19, "Syntax error: value, object or array expected."}},
              "* Line 1, Column 19\n  Syntax error: value, object or array "
              "expected.\n");
@@ -3223,7 +3223,7 @@ JSONTEST_FIXTURE_LOCAL(CharReaderTest, parseString) {
     bool ok = reader->parse(doc, doc + std::strlen(doc), &root, &errs);
     JSONTEST_ASSERT(ok);
     JSONTEST_ASSERT(errs.empty());
-    JSONTEST_ASSERT_EQUAL(u8"\u8A2a", root[0].asString()); // "訪"
+    JSONTEST_ASSERT_EQUAL(reinterpret_cast<const char*>(u8"\u8A2a"), root[0].asString()); // "訪"
   }
   {
     char const doc[] = R"([ "\uD801" ])";

--- a/src/test_lib_json/main.cpp
+++ b/src/test_lib_json/main.cpp
@@ -1993,7 +1993,8 @@ JSONTEST_FIXTURE_LOCAL(ValueTest, StaticString) {
 
 JSONTEST_FIXTURE_LOCAL(ValueTest, WideString) {
   // https://github.com/open-source-parsers/jsoncpp/issues/756
-  const std::string uni = reinterpret_cast<const char*>(u8"\u5f0f\uff0c\u8fdb"); // "式，进"
+  const std::string uni =
+      reinterpret_cast<const char*>(u8"\u5f0f\uff0c\u8fdb"); // "式，进"
   std::string styled;
   {
     Json::Value v;
@@ -3110,8 +3111,8 @@ JSONTEST_FIXTURE_LOCAL(ReaderTest, strictModeParseNumber) {
 
 JSONTEST_FIXTURE_LOCAL(ReaderTest, parseChineseWithOneError) {
   checkParse(reinterpret_cast<const char*>(R"({ "pr)"
-             u8"\u4f50\u85e4" // 佐藤
-             R"(erty" :: "value" })"),
+                                           u8"\u4f50\u85e4" // 佐藤
+                                           R"(erty" :: "value" })"),
              {{18, 19, "Syntax error: value, object or array expected."}},
              "* Line 1, Column 19\n  Syntax error: value, object or array "
              "expected.\n");
@@ -3223,7 +3224,8 @@ JSONTEST_FIXTURE_LOCAL(CharReaderTest, parseString) {
     bool ok = reader->parse(doc, doc + std::strlen(doc), &root, &errs);
     JSONTEST_ASSERT(ok);
     JSONTEST_ASSERT(errs.empty());
-    JSONTEST_ASSERT_EQUAL(reinterpret_cast<const char*>(u8"\u8A2a"), root[0].asString()); // "訪"
+    JSONTEST_ASSERT_EQUAL(reinterpret_cast<const char*>(u8"\u8A2a"),
+                          root[0].asString()); // "訪"
   }
   {
     char const doc[] = R"([ "\uD801" ])";

